src/box/box.cc

```diff
@@ -3473,12 +3478,41 @@ bootstrap_from_master(struct replica *master)
 	enum applier_state wait_state = replication_anon ?
 					APPLIER_FETCH_SNAPSHOT :
 					APPLIER_INITIAL_JOIN;
-	applier_resume_to_state(applier, wait_state, TIMEOUT_INFINITY);
+	try {
+		applier_resume_to_state(applier, wait_state,
+					TIMEOUT_INFINITY);
+	} catch (...) {
+		return -1;
+	}
+
+	try {
+		applier_resume_to_state(applier, APPLIER_CONTINUE_SNAPSHOT,
+					TIMEOUT_INFINITY);
+	} catch (...) {
+		return -1;
+	}
```

Comment:
As Sergey said in his previous comment, you don't need 2 resumes. You can just keep the last one:
```Diff
--- a/src/box/box.cc
+++ b/src/box/box.cc
@@ -3475,23 +3475,12 @@ bootstrap_from_master_begin(struct replica *master)
 	 */
 
 	assert(!tt_uuid_is_nil(&INSTANCE_UUID));
-	enum applier_state wait_state = replication_anon ?
-					APPLIER_FETCH_SNAPSHOT :
-					APPLIER_INITIAL_JOIN;
-	try {
-		applier_resume_to_state(applier, wait_state,
-					TIMEOUT_INFINITY);
-	} catch (...) {
-		return -1;
-	}
-
 	try {
 		applier_resume_to_state(applier, APPLIER_CONTINUE_SNAPSHOT,
 					TIMEOUT_INFINITY);
 	} catch (...) {
 		return -1;
 	}
-
 	return 0;
 }
```

Reply:
Done